src/proxy/outbound.rs

```diff
@@ -93,11 +93,11 @@ struct OutboundConnection {
 }
 
 impl OutboundConnection {
-    async fn proxy(&self, mut stream: TcpStream) -> Result<(), Error> {
+    async fn proxy(&self, mut stream: TcpStream) -> anyhow::Result<()> {
```

Comment:
IMO we should be using proper Error enums where feasible, not just untyped anyhow results. This can become a new enum `UnknownSource`

Reply:
Sure， can do it in following pr